libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -212,11 +233,7 @@ void ImmutableValidator::analyseVariableReference(VariableDeclaration const& _va
 				"Cannot write to immutable here: Immutable variables cannot be initialized inside a loop."
 			);
 		else if (m_inBranch)
-			m_errorReporter.typeError(
-				4599_error,
-				_expression.location(),
-				"Cannot write to immutable here: Immutable variables cannot be initialized inside an if statement."
-			);
+			m_immutableStateVariablesWrittenInBranch.emplace(&_variableReference);
```

Comment:
I think if this is not always an error case, we cannot continue with "else if".

Can you create a test case that detects this bug?

Reply:
Thanks for catching this; I think I have a better understanding of your earlier concerns now.

This PR essentially (1) relaxes the multiple write check to allow single write in each branch, and (2) checks for at least a single write in each branch. I made necessary changes for (1) and augmented test case to demonstrate that it falls back to other existing checks in relaxed cases.